packages/react/src/components/Forms/DatePicker/index.tsx

```diff
@@ -0,0 +1,134 @@
+import { Button } from ":/components/Button";
+import { Field, FieldProps } from ":/components/Forms/Field";
+import { LabelledBox } from ":/components/Forms/LabelledBox";
+import { DateField } from ":/components/Forms/DatePicker/DateField";
+import { Calendar } from ":/components/Forms/DatePicker/Calendar";
+
+const useHandleClickOutside = (
+  ref: RefObject<HTMLDivElement>,
+  onClose: any
+) => {
+  useEffect(() => {
+    const outsideListenerEvent = (event) => {
+      if (!ref.current || ref.current.contains(event.target)) {
+        return;
+      }
+      onClose();
+    };
+    document.addEventListener("click", outsideListenerEvent, true);
+    return () =>
+      document.removeEventListener("click", outsideListenerEvent, true);
+  }, []);
+};
+
+type DatePickerProps = PropsWithChildren &
+  FieldProps & {
+    label: string;
+    name?: string;
+    defaultValue?: CalendarDate;
+    minValue?: CalendarDate;
+    maxValue?: CalendarDate;
+    onChange?: any;
+    disabled?: boolean;
+    // todo : check with nathan why required attribute is not defined in the FieldProps
+  };
+
+export const DatePicker = ({
+  label,
+  name,
+  disabled = false,
+  ...props
+}: DatePickerProps) => {
+  const datePickerOptions: DatePickerStateOptions<DateValue> = {
+    ...props,
+    shouldCloseOnSelect: true,
+    granularity: "day",
+    isDisabled: disabled,
+  };
+
+  const state = useDatePickerState(datePickerOptions);
+  const pickerRef = useRef<HTMLDivElement>();
+  const wrapperRef = useRef<HTMLDivElement>();
+  useHandleClickOutside(pickerRef, state.close);
+
+  const { fieldProps, buttonProps, groupProps, calendarProps } = useDatePicker(
+    datePickerOptions,
+    state,
+    wrapperRef
+  );
+
+  const [labelAsPlaceholder, setLabelAsPlaceholder] = useState(true);
+  const [isDateInvalid, setIsDateInvalid] = useState(false);
+
+  useEffect(() => {
+    setLabelAsPlaceholder(!state.isOpen && !state.value);
+  }, [state.value, state.isOpen]);
+
+  useEffect(() => {
+    setIsDateInvalid(
+      state.validationState === "invalid" || props.state === "error"
+    );
+  }, [state.validationState, props.state]);
+
+  return (
+    <Field {...props}>
+      <div
+        ref={pickerRef}
+        className={classNames("c__date-picker", {
+          "c__date-picker--disabled": disabled,
+          "c__date-picker--invalid": isDateInvalid,
+          "c__date-picker--picked":
+            !isDateInvalid && !disabled && !labelAsPlaceholder,
+        })}
+      >
+        <div
+          className={classNames("c__date-picker__wrapper", {
+            "c__date-picker__wrapper--clickable": labelAsPlaceholder,
+          })}
+          ref={wrapperRef}
+          {...groupProps}
+          role="button"
+          tabIndex="0"
+          onClick={() => !state.isOpen && state.open()}
+        >
+          {state.value && (
+            <input type="hidden" name={name} value={state.value?.toString()} />
+          )}
+          <Button
```

Comment:
This button aria-label is "Calendrier" in french, but I think we might need to provide a more informative label like "Afficher le calendrier" or something like that, WDYT?

Reply:
I agree.
I would add that it's a toggle button. Then,  "Afficher" might be a bit misleading when the calendar is already opened? 
Would it better to make this label dynamic based on menu's state?